### PR TITLE
Rewrite docs (again!)

### PR DESCRIPTION
**Pull request by gaearon**
Another, hopefully final, stab at docs. Supersedes #289 and #140 and will gradually take some content from there. Uses Gitbook for rendering a static website which you can launch by running `npm run docs:watch`.

### CODE REVIEW

**Comment by gaearon** [comment]
Some parts I think are fairly good already:

https://github.com/gaearon/redux/blob/rewrite-docs-again/README.md
https://github.com/gaearon/redux/blob/rewrite-docs-again/docs/API%20Reference/createStore.md
https://github.com/gaearon/redux/blob/rewrite-docs-again/docs/API%20Reference/Store.md

**Comment by gaearon** [comment]
All right. You can take a look at preliminary 1.0 docs here: http://gaearon.github.io/redux

There's just a glossary and API reference there. “Getting Started” and “Recipes” coming later.
What do you think?

**Comment by iclanzan** [comment]
It is beautiful. Nice work!

**Comment by knowbody** [comment]
really good job @gaearon!

File: docs/api/Store.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+#### Returns
+
+(Object<sup>†</sup>): The dispatched action.
+
+#### Notes
+
+<sup>†</sup> The “vanilla” store implementation you get by calling [`createStore`](createStore.md) only supports plain object actions and hands them immediately to the reducer.
+
+However, if you wrap [`createStore`](createStore.md) with [`applyMiddleware`](applyMiddleware.md), the middleware can interpret actions differently, and provide support for dispathing [intents](../Glossary.md#intent). Intents are usually asynchronous primitives like Promises, Observables, or thunks.
```
Comment (dariusriggins) [inline docs/api/Store.md]: dispathing -> dispatching.

**Comment by demohi** [comment]
:+1:  @gaearon

**Comment by gaearon** [comment]
This is in `master` now, closing PR.

**Pull request closed without merging**